.github/actions/teams-message/action.yml

```diff
@@ -0,0 +1,91 @@
+name: Send a message to Teams
+
+inputs:
+  msg:
+    description: The teams message text
+  teams_url:
+    description: passthrough for secrets.TEAMS_CORSO_CI_WEBHOOK_URL
+    default: ${{ secrets.TEAMS_CORSO_CI_WEBHOOK_URL }}
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v3
+
+    - name: set github ref
+      shell: bash
+      run: |
+        echo "github_reference=${{ github.ref }}" >> $GITHUB_ENV
+
+    - name: trim github ref
+      shell: bash
+      run: |
+        echo "trimmed_ref=${github_reference#refs/}" >> $GITHUB_ENV
+
+    - name: build urls
+      shell: bash
+      run: |
+        echo "logurl=$(printf '<https://github.com/alcionai/corso/actions/runs/%s|[Action]>' ${{ github.run_id }})" >> $GITHUB_ENV
+        echo "commiturl=$(printf '<https://github.com/alcionai/corso/commit/%s|[Commit]>' ${{ github.sha }})" >> $GITHUB_ENV
+        echo "refurl=$(printf '<https://github.com/alcionai/corso/%s|[Ref]>' ${{ env.trimmed_ref }})" >> $GITHUB_ENV
+
+    - name: use url or blank val
+      shell: bash
+      run: |
+        echo "STEP=${{ env.trimmed_ref || '' }}" >> $GITHUB_ENV
+        echo "JOB=${{ github.job || '' }}" >> $GITHUB_ENV
+        echo "LOGS=${{ github.run_id && env.logurl || '-' }}" >> $GITHUB_ENV
+        echo "COMMIT=${{ github.sha && env.commiturl || '-' }}" >> $GITHUB_ENV
+        echo "REF=${{ env.trimmed_ref && env.refurl || '-' }}" >> $GITHUB_ENV
+
+    - name: Send JSON payload to Teams Webhook
+      run: |
```

Comment:
this might need `shell: bash` to properly handle the value interpolation.  Looks like the dynamic references are coming through as plaintext at this time.

Reply:
ACK - I haven't actually tested this yet in GH but let me add that just in case - no reason it shouldn't be there.